tajo-rpc/src/main/java/org/apache/tajo/rpc/AsyncRpcClient.java

```diff
@@ -113,16 +87,11 @@ public void close() {
   }
 
   private class ProxyRpcChannel implements RpcChannel {
-    private final ClientChannelInboundHandler handler;
-
-    public ProxyRpcChannel() {
-      this.handler = getChannel().pipeline()
-          .get(ClientChannelInboundHandler.class);
+    private ClientChannelInboundHandler handler;
 
-      if (handler == null) {
-        throw new IllegalArgumentException("Channel does not have " +
-            "proper handler");
-      }
+    private ClientChannelInboundHandler handler() {
+      return handler == null ? handler = getChannel().pipeline()
```

Comment:
The code part cannot be reached if this connection is not acquired, asserting the channel is established already. And callMethod() is already regarding getChannel() returns not-null value. I'll add RuntimeException on getChannel() if it can be null.